src/module-elasticsuite-thesaurus/Model/Index.php

```diff
@@ -275,17 +293,17 @@ private function getSynonymRewrites($storeId, $queryText, $type, $maxRewrites)
      */
     private function getQueryCombinations($storeId, $queryText)
     {
-        if (str_word_count($queryText) < 2) {
-            return [$queryText]; // No need to compute variations of shingles with a one-word-query.
-        }
-
         // Generate the shingles.
         // If we analyze "long sleeve dress", we'll obtain "long_sleeve", and "sleeve_dress".
+        // If stemming is enabled, analysis of "dresses" will give "dress"
+        // and analysis of "long sleeves" will give "long_sleev"
         // We'll also obtain the position (start_offset and end_offset) of those shingles in the original string.
         $indexName = $this->getIndexAlias($storeId);
         try {
+            // No need to compute variations of shingles with a one-word-query but we need to extract the stem if enabled.
+            $analyzer = str_word_count($queryText) < 2 ? 'clean' : 'shingles';
```

Comment:
Are you sure there is no side effects of using the "clean" analyzer if the stemming in thesaurus feature is disabled ?

Reply:
I did not see any side effects, but yes this step is useless if thesaurus stemming is disabled.
I re-add the test to by pass this step